grpc/server.go

```diff
@@ -69,10 +69,16 @@ func NewServer(c *cmd.GRPCServerConfig, tlsConfig *tls.Config, statsRegistry pro
 	if tlsConfig == nil {
 		return nil, nil, errNilTLS
 	}
+
 	acceptedSANs := make(map[string]struct{})
 	for _, name := range c.ClientNames {
 		acceptedSANs[name] = struct{}{}
 	}
+	for _, service := range c.Services {
+		for _, name := range service.ClientNames {
+			acceptedSANs[name] = struct{}{}
+		}
+	}
```

Comment:
Oops, in my comment here I was working off an outdated model of how we build gRPC servers. It looks like the current `Setup` approach with a callback could suit this check pretty well. I'll comment on #6482.

Reply:
I've done #1, but as you noted here numbers 2 and 3 will have to be incorporated into the second PR in this stack, if that's okay with you?

Reply:
Actually wait, doing #1 this is more complex than I thought.

In particular: the Remote VAs have one allowed client for their VA service (the VA itself) but zero allowed clients for their CAA service (because we only do single-perspective CAA). So the allowed clients list for that service is empty. And in the current architecture, we can't just say "don't register the CAA service on remote VAs", because remote VAs don't actually know that they're remote. There's no config value that unambiguously indicates that.